src/components/ListHint.tsx

```diff
@@ -0,0 +1,203 @@
+import {
+  Badge,
+  Box,
+  Center,
+  Flex,
+  FormControl,
+  FormErrorMessage,
+  Heading,
+  HStack,
+  Input,
+  Stack,
+  Text,
+  VStack,
+  useToast,
+  CloseButton,
+} from "@chakra-ui/react";
+import React, { useEffect, useState } from "react";
+import { AiOutlineArrowRight } from "react-icons/ai";
+import { BsCircleFill } from "react-icons/bs";
+import AppLayout from "../layouts/AppLayout";
+import { checkError, hints } from "../pages/Setup";
+import BoxContainer from "./BoxContainer";
+import PrimaryButton from "./PrimaryButton";
+
+function HintCard({
+  text,
+  index,
+  onEdit,
+}: {
+  text: string;
+  index: number;
+  onEdit: (index: number) => void;
+}) {
+  return (
+    <Box
+      maxW="container.md"
+      width={{ base: "100%", sm: "80%" }}
+      boxShadow="md"
+      p="4"
+      rounded="3xl"
+      bg="white"
+      cursor="pointer"
+      transition={"all 0.2s ease-in-out"}
+      _hover={{ bg: "pink.50", transform: "scale(1.05)" }}
+      onClick={() => {
+        onEdit(index);
+      }}
+    >
+      <Flex alignItems="center" justifyContent="space-between" gap="2">
+        <Flex gap="5" alignItems="center">
+          <Box bg="#EBDCFC" p="5" rounded="lg"></Box>
+          <VStack alignItems="flex-start">
+            <Heading as="h3" size="sm">
+              Hint {index + 1}
+            </Heading>
+            <HStack>
+              <Text fontSize="8" color="gray.200">
+                <BsCircleFill />
+              </Text>
+              <Text>{text}</Text>
+            </HStack>
+          </VStack>
+        </Flex>
+      </Flex>
+    </Box>
+  );
+}
+
+function ListHint({
+  hints,
+  setHints,
+}: {
+  hints: string[];
+  setHints: React.Dispatch<React.SetStateAction<hints>>;
+}) {
+  const [indexHint, setIndexHint] = useState(-1);
+  const [input, setInput] = useState("");
+  const onEdit = (index: number) => {
+    setIndexHint(index);
+  };
+  const SubmitHints = () => {
+    console.log(hints);
+  };
+  const toast = useToast();
+  const toastError = () => {
+    if (checkError(input, hints)) {
+      toast.closeAll();
+      toast({
+        position: "top",
+        title: "Fail",
+        status: "error",
+        variant: "left-accent",
+        isClosable: true,
+        duration: 3000,
+        description: checkError(input, hints),
+      });
+    }
+  };
+  useEffect(() => {
+    if (indexHint !== -1) {
```

Comment:
could be combined into `onEdit`

Reply:
if combined to onEdit and remove useEffect, last text that user type a hint in that index of array hint will disappear;-;  (What should I do to fix this?;-;)